File: task_equivalents/llm_translations/gpt/optimized/r/data_visualization/02_distribution_boxplot.R
library(readr)

college <- read_csv("data/csv/College.csv", show_col_types = FALSE)

rownames(college) <- college[[1]]
college <- college[, -1, drop = FALSE]

college$Private <- as.factor(college$Private)
college$Elite <- factor(ifelse(college$Top10perc > 50, "Yes", "No"))

boxplot(Outstate ~ Elite, data = college, xlab = "Elite", ylab = "Outstate")

op <- par(mfrow = c(2, 2))
hist(college$Apps, breaks = 50, xlab = "new applications", main = "")
hist(college$Enroll, breaks = 45, xlab = "new enrollment", main = "")
hist(college$Expend, breaks = 30, xlab = "Instructional expenditure per student", main = "")
hist(college$Outstate, xlab = "Out-of-state tuition", main = "")
par(op)

File: task_equivalents/llm_translations/gpt/optimized/python/data_visualization/02_distribution_boxplot.py
import pandas as pd
import matplotlib.pyplot as plt

college = pd.read_csv("data/csv/College.csv")

college.index = college.iloc[:, 0]
college = college.iloc[:, 1:].copy()

college["Elite"] = pd.Categorical(
    pd.Series("No", index=college.index).where(college["Top10perc"] <= 50, "Yes")
)

plt.figure()
plt.boxplot(
    [
        college.loc[college["Elite"] == "No", "Outstate"],
        college.loc[college["Elite"] == "Yes", "Outstate"],
    ],
    tick_labels=["No", "Yes"],
)
plt.xlabel("Elite")
plt.ylabel("Out-of-state tuition")

fig, axes = plt.subplots(2, 2)
axes[0, 0].hist(college["Apps"], bins=50)
axes[0, 0].set_xlabel("new applications")
axes[0, 0].set_title("")

axes[0, 1].hist(college["Enroll"], bins=45)
axes[0, 1].set_xlabel("new enrollment")
axes[0, 1].set_title("")

axes[1, 0].hist(college["Expend"], bins=30)
axes[1, 0].set_xlabel("Instructional expenditure per student")
axes[1, 0].set_title("")

axes[1, 1].hist(college["Outstate"])
axes[1, 1].set_xlabel("Out-of-state tuition")
axes[1, 1].set_title("")

plt.tight_layout()
plt.show()